### PR TITLE
[Beam-5243] Redo of buggy PR #6903

### PR DESCRIPTION
**Pull request by robinyqiu**
The original PR #6903 cause a bug so it was reverted by PR #7048. This is a redo of PR #6903 with a correct implementation.
(This is a quick redo. I am still looking for a more elegant solution to this problem at the Gradle level, without the need to create the run_tox.sh script wrapper.)

r: @aaltay @udim 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by aaltay** [comment]
@udim could you make the first review? Does this fix the issue you found?

File: sdks/python/scripts/run_tox.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+fi
+
+tox -c tox.ini --recreate -e $1
+exit_code=$?
+# Retry once for the specific exit code -11.
+if [[ $exit_code == -11 ]]; then
+  tox -c tox.ini --recreate -e $1
+  exit_code=$?
+fi
+exit $exit_code
```
Comment (udim) [inline sdks/python/scripts/run_tox.sh]: nit: missing newline at EOF

Comment (robinyqiu) [reply sdks/python/scripts/run_tox.sh]: Fixed

**Comment by udim** [comment]
I believe the script isn't catching the right exit code:
https://builds.apache.org/job/beam_PreCommit_Python_Commit/2659/consoleFull

I think the actual exit code to check is 245:
```sh
$ bash -c 'exit -11'; echo $?
245
```

**Comment by robinyqiu** [comment]
Sent out a PR to fix this: #7131

**Pull request merged**